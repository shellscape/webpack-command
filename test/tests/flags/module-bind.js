const { apply, build, crcDist, distContains, test, validate } = require('../../util');

test('--module-bind', module, () => {
  const fixture = `module-bind/bind`;
  const opts = { fixture };

  let config;

  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should bind and build`, () =>
    build(config).then((result) => {
      expect(distContains('loader: module.exports')).toBe(true);
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});

test('--module-bind-pre', module, () => {
  const fixture = `module-bind/bind-pre`;
  const opts = { fixture };

  let config;

  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should bind and build`, () =>
    build(config).then((result) => {
      expect(distContains(`pre = true`)).toBe(true);
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});

test('--module-bind-post', module, () => {
  const fixture = `module-bind/bind-post`;
  const opts = { fixture };

  let config;

  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should bind and build`, () =>
    build(config).then((result) => {
      expect(distContains(`post = true`)).toBe(true);
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});

test('--module-bind-all', module, () => {
  const fixture = `module-bind/bind-all`;
  const opts = { fixture };

  let config;

  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should bind and build`, () =>
    build(config).then((result) => {
      expect(distContains(`pre = true`)).toBe(true);
      expect(distContains(`post = true`)).toBe(true);
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});
