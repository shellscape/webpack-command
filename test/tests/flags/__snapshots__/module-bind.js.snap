// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --module-bind > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": undefined,
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind > should bind and build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  3.93 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 48 bytes {main} [built]"
`;

exports[`Flags > --module-bind > should bind and build #1 1`] = `"e9ce1d9a"`;

exports[`Flags > --module-bind-all > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": undefined,
        "loader": "json-loader",
        "test": /\\\\\\.json\\$/,
      },
      Object {
        "enforce": "pre",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-pre",
        "test": /\\\\\\.js\\$/,
      },
      Object {
        "enforce": "post",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-post",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind-all > should bind and build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  3.96 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 77 bytes {main} [built]"
`;

exports[`Flags > --module-bind-all > should bind and build #1 1`] = `"96720baf"`;

exports[`Flags > --module-bind-post > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": "post",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-post",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind-post > should bind and build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  3.94 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 59 bytes {main} [built]"
`;

exports[`Flags > --module-bind-post > should bind and build #1 1`] = `"f36cb12f"`;

exports[`Flags > --module-bind-pre > should apply #0 1`] = `
Object {
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": "pre",
        "loader": "<PROJECT_ROOT>/test/fixtures/common/loader-pre",
        "test": /\\\\\\.js\\$/,
      },
    ],
  },
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --module-bind-pre > should bind and build #0 1`] = `
"Version: webpack 4.39.2
  Asset      Size  Chunks             Chunk Names
main.js  3.94 KiB    main  [emitted]  main
Entrypoint main = main.js
[./test/fixtures/common/entry-a.js] 58 bytes {main} [built]"
`;

exports[`Flags > --module-bind-pre > should bind and build #1 1`] = `"41753a3b"`;
